src/RecurringPurchase.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.23;
+
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {IERC20Metadata} from "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+
+import {AggregatorV3Interface} from "src/vendor/chainlink/AggregatorV3Interface.sol";
+import {ISwapRouter} from "v3-periphery/interfaces/ISwapRouter.sol";
+
+import {QuarkWallet} from "quark-core/src/QuarkWallet.sol";
+import {QuarkScript} from "quark-core/src/QuarkScript.sol";
+
+/**
+ * @title Recurring Purchase Script
+ * @notice Quark script that performs a swap on a regular interval.
+ * @author Legend Labs, Inc.
+ */
+contract RecurringPurchase is QuarkScript {
+    using SafeERC20 for IERC20;
+
+    error BadPrice();
+    error InvalidInput();
+    error PurchaseConditionNotMet();
+
+    /// @notice Emitted when a swap is executed
+    event SwapExecuted(
+        address indexed sender,
+        address indexed recipient,
+        address tokenIn,
+        address tokenOut,
+        uint256 amountIn,
+        uint256 amountOut,
+        bytes path
+    );
+
+    /// @notice The base slippage factor where `1e18` represents 100% slippage tolerance
+    uint256 public constant BASE_SLIPPAGE_FACTOR = 1e18;
+
+    /**
+     * @dev Note: This script uses the following storage layout in the QuarkStateManager:
+     *         mapping(bytes32 hashedPurchaseConfig => uint256 nextPurchaseTime)
+     *             where hashedPurchaseConfig = keccak256(PurchaseConfig)
+     */
+
+    /// @notice Parameters for a recurring purchase order
+    struct PurchaseConfig {
+        uint256 interval;
```

Comment:
unit? also presumably seconds

might consider being explicit there: `intervalSeconds` is not too cumbersome.

Reply:
Yeah it's in seconds. I usually would attach units to these, but in Solidity I think it's always assumed to be in seconds unless otherwise stated. For example, `expiry` is used widely without specifying seconds. I'll leave a `@dev` comment to make sure there's no confusion.

Reply:
comment is totallly sufficient 👍